src/twisted/web/http.py

```diff
@@ -2553,6 +2553,12 @@ def allContentReceived(self):
         req = self.requests[-1]
         req.requestReceived(command, path, version)
 
+    def upgradeToProtocol(self, protocol: IProtocol) -> None:
```

Comment:
I think that this needs a docstring.

Also, maybe we should make it private and only our `wsproto` adapter code should use it for now.

```suggestion
    def upgradeToProtocol(self, protocol: IProtocol) -> None:
```

Reply:
Good catch; we definitely should not have new, public APIs with no direct tests of their own and no documentation.